src/shims/fs.rs

```diff
@@ -148,13 +152,24 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
         if let Some(handle) = this.machine.file_handler.handles.remove(&fd) {
-            // `File::sync_all` does the checks that are done when closing a file. We do this to
-            // to handle possible errors correctly.
-            let result = this.try_unwrap_io_result(handle.file.sync_all().map(|_| 0i32));
-            // Now we actually close the file.
-            drop(handle);
-            // And return the result.
-            result
+            // We sync the file if it was opened in a mode different than read-only.
+            if !handle.read_only {
```

Comment:
Unless there are good arguments for the opposite, it's better to put the positive case first (`if handle.read_only`). That avoids a mental double-negation when considering the `else` branch.

Reply:
I did it this way to keep the "usual" behavior first and then be able to explain why this exception is required.

Reply:
Why is read-write more "usual" than read-only?

Reply:
huh that's a good question. I don't think that's the case. Taking that into account, should I change the field inside the handle to read-write then?

Reply:
Whatever you prefer, as long as you avoid the negation here. ;)

Reply:
I changed it to avoid the double negation.